tools/linter/adapters/ruff_linter.py

```diff
@@ -0,0 +1,462 @@
+"""Adapter for https://github.com/charliermarsh/ruff."""
+
+from __future__ import annotations
```

Comment:
Is this really needed?
```suggestion
```

Reply:
I tried to remove it. everything broke. Looks like something was relying on its existence in this module

Reply:
This is needed for python 3.8 compatibility for the annotations eg list[…]